numpy/core/shape_base.py

```diff
@@ -449,7 +450,7 @@ def _block_format_index(index):
     return 'arrays' + idx_str
 
 
-def _block_check_depths_match(arrays, parent_index=[]):
```

Comment:
A better pattern for avoiding this trap is to use `()` instead of `[]` and `mappingproxy()` instead of `{}`.

Using that approach ensures that no mutation can ever occur.